go/worker/lambda/lambdaFunction.go

```diff
@@ -371,19 +369,19 @@ func (f *LambdaFunc) Task() {
 
 		// kill or start at most one instance to get closer to
 		// desired number
-		if f.instances.Len() < desiredInstances {
-			f.printf("increase instances to %d", f.instances.Len()+1)
+		if f.nInstances < desiredInstances {
+			f.printf("increase instances to %d", f.nInstances+1)
 			f.newInstance()
 			lastScaling = &now
-		} else if f.instances.Len() > desiredInstances {
-			f.printf("reduce instances to %d", f.instances.Len()-1)
-			waitChan := f.instances.Back().Value.(*LambdaInstance).AsyncKill()
```

Comment:
This was the problematic line -- we chose the most recently spun up LambdaInstance, which need not be the idle one.